### DISCUSSION TITLE
__clone() method removal from Singleton

### DISCUSSION THREAD

**Discussion by AliN11** [DesignPatternsPHP/DesignPatternsPHP#434] [General]
Hello. In the last commit the `__clone()` method was removed from the Singleton pattern. While defining this method as private guarantees that only one single instance will exist, can you explain the removal? Thanks.

```php
<?php

$obj = Singleton::getInstance();
$obj2 = clone Singleton::getInstance();

var_dump($obj === $obj2); // false
```

**Comment by domnikl** [comment]
Hi @AliN11, you are right, that method should not be removed. Seems like rector did remove it while running the refactorings towards PHP8 and I didn't notice. My bad, will re-add that.

**Reply by AliN11** [reply]
Aha. I created a pull request #435
